amida-tech/saraswati-cql-execution
# Diabetes Shared Library

# Related Tickets



- [SAR-97](https://amida.atlassian.net/browse/SAR-97)

# How Things Worked (or Didn't) Before This PR


- Diabetes BP cql and Diabetes A1C share a lot of the same logic and are in different files.

# How Things Work Now (And How to Test)



- There is a common library that contains the logic and is used by both cql files for BP and A1C. The expressions remain the same since we need them for saraswati reports.
### How to Test
- Checkout this branch
- Run `Yarn Start`
- In another terminal run `node receiver.js `
- Then modify (ie. add a space and save) the `cdc_hba1c-patients.json` data file and then the `cdc_diabetes-bp-patients.json` data file. The results should be exactly the same as if you where to run this same steps in the `develop` branch
